### PR TITLE
feat: add typescript server example

### PR DESCRIPTION
**Pull request by hardfist**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by hardfist** [comment]
moving  https://github.com/fastify/fastify/pull/1815 here
fix https://github.com/fastify/fastify/issues/1349

File: typescript-server/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        type: 'object',
+        properties: {
+          hello: {
+            type: 'string'
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (mcollina) [inline typescript-server/src/app.ts:25]: Would you mind using https://github.com/fastify/fluent-schema? It feels more natural!

File: typescript-server/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "ts-node": "^8.3.0",
+    "typescript": "^3.5.3"
+  },
+  "dependencies": {
+    "@types/cors": "^2.8.6",
+    "@types/node": "^12.7.2",
+    "cors": "^2.8.5",
```
Comment (mcollina) [inline typescript-server/package.json]: I'm not sure you are using cors, are you?

File: typescript-server/src/app.ts (hunk 1 above)
Comment (hardfist) [reply typescript-server/src/app.ts:25]: ok, I just  tried this, but met some problem , can you give some example to reference

File: typescript-server/package.json (hunk 2 above)
Comment (hardfist) [reply typescript-server/package.json]: fix now

**Pull request closed without merging**